File: cpp/stackimplementation.cpp
#include<bits/stdc++.h>
using namespace std;
class A
{
   
    int* arr;
    int n;
    int top;
    public:
    A()
    {
        arr=new int[n];
        top=-1;
    }

void push(int val)
{
    if(top==(n-1))
    {
        cout<<"STACK OVERFLOWED"<<endl;
        return;
    }
    top=top+1;
    arr[top]=val;
    
}
void pop()
{
    if(top==-1)
    {
        cout<<"STACK UNDERFLOW:"<<endl;
    }
    top=top-1;
}
int Top()
{
    if(top==-1)
    {
        cout<<"STACK IS EMPTY"<<endl;
        return -1;
    }
    return arr[top];
}
bool empty()
{
    if(top==-1)
    return true;
    else 
    return false;
}
};
int main()
{
    A st;
    int n;
    cin>>n;
   

    for(int i=0;i<n;i++)
    {
        int element;
        cin>>element;
        st.push(element);
    }
    cout<<st.Top()<<endl;
    st.pop();
    cout<<st.Top()<<endl;
    
}

File: Java/stack_implementation.java
/**
 * stackmain
 */



class stack{
    int top=-1;
    int limit = 5;
    float array[] = new float[limit];

    boolean push(float x)
    {
        if (top >= (limit - 1)) {
            System.out.println("Stack Overflow");
            return false;
        }
        else {
        array[++top] = x;
            return true;
        }
    }

    float pop()
    {
        if (top < 0) {
            System.out.println("Stack Underflow");
            return 0;
        }
        else {
            float x = array[top--];
            return x;
        }
    }

    void display() {
        for(int i=top;i>=0;i--) {
            System.out.println("| " + array[i] + " ");
        }
        System.out.println("\n");
    }

    public static void copy(stack s1, stack s2){
        int i;
        for(i=0;i<=s1.top;i++) {
            s2.array[i] = s1.array[i];
        }
        s2.top = s1.top;
    }
 
}


 class stack_implementation {




    public static void main(String[] args) {
        stack stack1 = new stack();
        stack stack2 = new stack();
        stack1.push(2.1f);
        stack1.push(2.2f);
        stack1.push(2.3f);
        stack1.pop();
        stack1.push(2.4f);
        stack1.pop();
        stack.copy(stack1,stack2);
        stack1.display();
        stack2.display();
    }
}